bin/upload_metric.py

```diff
@@ -36,12 +37,20 @@ def main(args=None):
     log_level = ArConfig.get('logging', 'log_level')
     log = init_log(log_mode, log_file, log_level, 'argo.upload_metric')
 
-    # call prefilter
-    cmd_pref = [os.path.join(arsync_exec, 'prefilter-avro'), '-d', args.date]
-
-    log.info("Calling prefilter-avro for date:%s", args.date)
-
-    run_cmd(cmd_pref, log)
+    # call prefilter if necessary for specified tenant
+    if ArConfig.has_option('jobs', arg_parser.parse_args().tenant + '_prefilter'):
```

Comment:
The potential bug I mention in the PR comment is here (in this conditional). So where does this come from?

The prefilter has been renamed (rationally I believe) from `prefilter-avro` to `prefilter-egi.py` in the latest agro-connectors version. I didn't want to write down anything tenant specific within the CE code, hence I made this configurable from within the configuration file. However, if a prefilter script is not provided then the `local_prefilter` variable will not be initialized (note it is required in line 67). As it stands it works, however, I am making the point now because in a future tenant where no prefilter is given the thing will break.